src/watertap_contrib/seto/property_models/basic_water_properties.py

```diff
@@ -0,0 +1,397 @@
+                    hold_state=True.
+            outlvl : sets output level of of logging
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+        init_log.info("State Released.")
+
+
+@declare_process_block_class("BasicWaterStateBlock", block_class=_BasicWaterStateBlock)
+class BasicWaterStateBlockData(StateBlockData):
+    """
+    General purpose StateBlock for Zero-Order unit models.
+    """
+
+    def build(self):
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.flow_vol = Var(
+            initialize=1,
+            domain=PositiveReals,
+            doc="Volumetric flow rate",
+            units=pyunits.m**3 / pyunits.s,
+        )
+
+        self.conc_mass_comp = Var(
+            self.params.solute_set,
+            initialize=1,
+            domain=PositiveReals,
+            doc="Mass concentration of each solute",
+            units=pyunits.kg / pyunits.m**3,
+        )
+
+    # Other properties
+    def _flow_mass_comp(self):
+
+        self.flow_mass_comp = Var(
+            self.params.component_list,
+            initialize=1,
+            domain=PositiveReals,
+            doc="Mass flowrate of each component",
+            units=pyunits.kg / pyunits.s,
+        )
+
+        def rule_flow_mass_comp(b, j):
+            if j == "H2O":
+                return b.flow_mass_comp[j] == b.flow_vol * b.dens_mass
+            else:
+                return b.flow_mass_comp[j] == b.flow_vol * b.conc_mass_comp[j]
+
+        self.eq_flow_mass_comp = Constraint(
+            self.params.component_list, rule=rule_flow_mass_comp
+        )
+
+    def _temperature(self):
+        self.temperature = Var(
+            initialize=298.15,
+            bounds=(273.15, 373.15),
+            units=pyunits.K,
+            doc="Temperature",
+        )
+
+    def _pressure(self):
+        self.pressure = Var(
+            initialize=101325,
+            bounds=(1e5, None),
+            units=pyunits.Pa,
+            doc="Pressure",
+        )
+
+    def _dens_mass(self):
+        self.dens_mass = Param(
+            initialize=self.params.dens_mass_default,
+            units=pyunits.kg / pyunits.m**3,
+            mutable=True,
+            doc="Mass density of flow",
+        )
+
+    def _visc_d(self):
+        self.visc_d = Param(
+            initialize=self.params.visc_d_default,
+            units=pyunits.kg / pyunits.m / pyunits.s,
+            mutable=True,
+            doc="Dynamic viscosity of solution",
+        )
+
+    def get_material_flow_terms(blk, p, j):
+        return blk.flow_mass_comp[j]
+
+    def get_enthalpy_flow_terms(blk, p):
+        raise NotImplementedError
+
+    def get_material_density_terms(blk, p, j):
+        return blk.conc_mass_comp[j]
```

Comment:
I suspect an error here if you add a test for this and try to get material density for water. Maybe you should add a conditional here to return density if j is H2O.